feat(storage): support bucket restore in soft delete phase 2

mahendra-google opened the pull request:
Feature request includes the following:-
1. Get Bucket Generation
2. Get a Soft Deleted Bucket (Also soft-delete time and hard-delete time)
3. List Soft Deleted Buckets
4. Restore a Soft Deleted Bucket


Includes
**Rebase PR on upstream/main (same as origin/main here ) with squashed reviewed commits till now**

mahendra-google committed:
Co-authored-by: Amanda Tarafa Mas <14878252+amanda-tarafa@users.noreply.github.com>

Co-authored-by: Amanda Tarafa Mas <14878252+amanda-tarafa@users.noreply.github.com>

	modified:   generator-input/apis.json

thiyaguk09 reviewed (approved):
LGTM

mahendra-google committed:
file-scoped namespace and generate bucket name method added

mahendra-google committed:
Adding all tests in single test , addition of generatebucket name method

mahendra-google committed:
adding a blank line between one property and the next

mahendra-google committed:
List Soft Deleted Bucket test case changes

mahendra-google committed:
Removal of encryption key support for restore bucket request and removal of package version override

thiyaguk09 commented on apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.IntegrationTests/RestoreBucketTest.cs:
```diff
@@ -16,28 +16,27 @@
-            Assert.Equal(bucket.Name, restored.Name);
-            Assert.Equal(bucket.Generation, restored.Generation);
-        }
+        var restored = await _fixture.Client.RestoreBucketAsync(softDeleteBucket.Name, softDeleteBucket.Generation.Value);
```
Could we consider using "restoredBucket" instead of "restored"?

mahendra-google replied:
I changed variable name to a more meaningful name

thiyaguk09 commented on apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.IntegrationTests/ListBucketsTest.cs:
```diff
@@ -78,19 +78,19 @@ public void PartialResponses()
             {
-                // Check if list contains only soft deleted buckets created in storage fixture
-                if (bucket.Name.EndsWith("soft-delete"))
+                if (bucket.Name == softDeleteBucket.Name)
```
Actually, it will return only the soft-deleted bucket list, so why is this bucket name comparison needed? Simply asserting the entire array without the if condition would be better.

mahendra-google replied:
I have made suggested changes. Please review and let me know.

mahendra-google committed:
Recommended changes as per internal PR review

thiyaguk09 reviewed (approved):
LGTM, @shubhangi-google Approved, Further Review Requested

gcf-owl-bot[bot] committed:
fix: add oauth scopes to Places proto

PiperOrigin-RevId: 720741557

Source-Link: https://github.com/googleapis/googleapis/commit/280725e991516d4a0f136268faf5aa6d32d21b54

Source-Link: https://github.com/googleapis/googleapis-gen/commit/194c4c32941fe559dd1a9848f4891db595562af5
Copy-Tag: eyJwIjoiYXBpcy9Hb29nbGUuTWFwcy5QbGFjZXMuVjEvLk93bEJvdC55YW1sIiwiaCI6IjE5NGM0YzMyOTQxZmU1NTlkZDFhOTg0OGY0ODkxZGI1OTU1NjJhZjUifQ==

mahendra-google committed:
named arguments added while calling create bucket method

Co-authored-by: Amanda Tarafa Mas <14878252+amanda-tarafa@users.noreply.github.com>

mahendra-google committed:
Update apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/GetBucketOptionsTest.cs

Co-authored-by: Amanda Tarafa Mas <14878252+amanda-tarafa@users.noreply.github.com>

mahendra-google committed:
Update apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/GetBucketOptions.cs

Co-authored-by: Amanda Tarafa Mas <14878252+amanda-tarafa@users.noreply.github.com>

mahendra-google committed:
Update apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/ListBucketsOptionsTest.cs

Co-authored-by: Amanda Tarafa Mas <14878252+amanda-tarafa@users.noreply.github.com>

mahendra-google committed:
restore white space in Google.Cloud.Storage.V1.csproj

mahendra-google committed:
Update GetBucketOptionsTest.cs

mahendra-google committed:
Update ListBucketsTest.cs

mahendra-google committed:
Update GetBucketTest.cs

mahendra-google committed:
encryption key removed from RestoreBucketOptions

mahendra-google committed:
Initial commit for Assert.Contains and Assert.All in list bucket test

mahendra-google committed:
listbucket linters changes

mahendra-google committed:
restore bucket linter changes

mahendra-google committed:
getbucket linter changes

mahendra-google committed:
getbucket linter changes

mahendra-google committed:
restore bucket linter changes

mahendra-google committed:
listbucket linter changes

mahendra-google committed:
blank line added in getbucketest

mahendra-google committed:
linter changes as per editorconfig file rules

mahendra-google commented:
@amrik-mahal Internal PR is ready for re-review.

amanda-tarafa committed:
Apply suggestions from code review

amanda-tarafa committed:
Update apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.IntegrationTests/ListBucketsTest.cs
